File: programmers/연습문제/정렬/K번째수.js
function solution(array, commands) {
    let answer = [];
    for (let c = 0; c < commands.length; c ++) {
        // 배열 array의 i번째부터 j번째 숫자까지 자르기
        let cut_array = array.slice(commands[c][0] - 1, commands[c][1])
        // 자른 배열 정렬하기
        cut_array.sort(function(a, b) {
            return a - b
        })
        // 정렬 후 k번째 있는 수 추가하기
        answer.push(cut_array[commands[c][2] - 1])
    }
    return answer;
}


// let array = [1, 5, 2, 6, 3, 7, 4] 
// let commands = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]
// console.log(solution(array, commands))

File: programmers/연습문제/정렬/K번째수.kt
import java.util.*

val inputArray = intArrayOf(1, 5, 2, 6, 3, 7, 4)
val input1 = intArrayOf(2, 5, 3)
val input2 = intArrayOf(4, 4, 1)
val input3 = intArrayOf(1, 7, 3)
val inputCommand: Array<IntArray> = arrayOf(input1, input2, input3)


// solution 1
fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
    var answer = intArrayOf()
    for (command in commands) {
        val sliceArray = array.slice(command[0] - 1..command[1] - 1)
        answer += sliceArray.sorted()[command[2] - 1]
    }

    return answer
}

// solution 2 (개선 코드)
fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
    return commands.map { command ->
        array.slice(command[0] - 1..command[1] - 1).sorted()[command[2] - 1]
    }.toIntArray()
}


fun main() {
    print(solution(inputArray, inputCommand).contentToString())
}
